src/System.Runtime.Extensions/src/System/Net/WebUtility.cs

```diff
@@ -357,6 +358,93 @@ private static byte[] UrlEncode(byte[] bytes, int offset, int count)
             return expandedBytes;
         }
 
+        private static unsafe string UrlEncode(string originalString, char* s, int length)
+        {
+            int numSpaces = 0;
+            int numUnsafe = 0;
+            int numSafe = 0;
```

Comment:
Unfortunately not, as we want to keep track of the number of spaces, as spaces have to be escaped to '+'